### DISCUSSION TITLE
Switch to OAuth2.0

### DISCUSSION THREAD

**Discussion by Solomon1732** [streetcomplete/StreetComplete#4301] [Ideas]
OSM is [supporting](https://wiki.openstreetmap.org/wiki/OAuth) version 2.0; version 1.0a is considered legacy. The only place I see two versions being mentioned is in #1161 in [this](https://github.com/streetcomplete/StreetComplete/issues/1161#issuecomment-418407668) comment.

**Comment by matkoniecz** [comment]
Can you link any info that 1.0a is deprecated and should be avoided?

I remember that there was something complicated about this

**Reply by Solomon1732** [reply]
> This specification was obsoleted by [RFC 6749: The OAuth 2.0 Authorization Framework](https://tools.ietf.org/html/rfc6749). Implementers should use [RFC 6749](https://tools.ietf.org/html/rfc6749) instead of this specification.

https://oauth.net/core/1.0a/
> OpenStreetMap supports OAuth 1.0a and 2.0. OAuth 2.0 should be used for any new application.

https://wiki.openstreetmap.org/wiki/OAuth

**Reply by matkoniecz** [reply]
>  any new application

Is there any indication made by sysadmins/maintainers of OSM website that existing software should be rewritten?

(StreetComplete uses plenty of deprecated things, for example in Android realm: and many are not worth spending effort to replace or are impossible to replace for now)

**Reply by Solomon1732** [reply]
Not that I can tell

**Reply by mmd-osm** [reply]
See https://github.com/openstreetmap/openstreetmap-website/issues/3299#issuecomment-901907857

So yes, you should start looking into OAuth 2.0 in the near future. No exact dates, but there are for sure plans to retire OAuth 1.0a at one point.

By the way, Basic Authentication is another candidate on the list for removal at some point in the future (also no exact dates defined yet).

**Reply by matkoniecz** [reply]
@Solomon1732 Can you tell me how you checked that SC is using OAuth1?

**Reply by mmd-osm** [reply]
SC uses https://github.com/westnordost/osmapi, which depends on https://github.com/mttkay/signpost, implementing OAuth 1.0a. 

Looking at https://www.openstreetmap.org/user/{{your osm display_name}}/oauth_clients you should see StreetComplete underneath "My Authorised Applications". If you're the owner of client id and secret for SC, the SC application will also show up underneath  My Client Applications. This page is for OAuth1.0a client applications.

**Comment by FloEdelmann** [comment]
Cross reference:

* #5322

**Reply by matkoniecz** [reply]
And https://github.com/westnordost/osmapi/issues/33 which seems better issue for that
